src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseService.java

```diff
@@ -264,13 +269,20 @@ public ProgrammingExercise createProgrammingExercise(ProgrammingExercise program
         // See https://github.com/ls1intum/Artemis/pull/7451/files#r1459228917
         programmingExercise.setSolutionParticipation(null);
         programmingExercise.setTemplateParticipation(null);
+        if (isImportedFromFile) {
+            // We have to set the id to null in order to generate a new id for the build config
+            programmingExercise.getBuildConfig().setId(null);
+        }
```

Comment:
Why only when importing from file? Do you want to reuse the same object when importing between courses? I don't think this should be the case, changes in one exercise should not be visible in the other exercise.